src/data/nat/modeq.lean

```diff
@@ -389,3 +389,96 @@ lemma rotate_one_eq_self_iff_eq_repeat [nonempty α] {l : list α} :
     λ h, rotate_eq_self_iff_eq_repeat.mpr h 1⟩
 
 end list
+
+--------------------------------------------------------------------------------------------------
+--------------------------------------------------------------------------------------------------
+--  PR THIS vvvv  --------------------------------------------------------------------------------
+--------------------------------------------------------------------------------------------------
+--------------------------------------------------------------------------------------------------
+
+namespace nat
+
+variables {a b c d m n : ℕ}
+
+-- Apostol, Theorem 5.3
+lemma mul_left_iff (c : ℕ) (hc : 0 < c) : a ≡ b [MOD m] ↔ c * a ≡ c * b [MOD c * m] :=
+begin
+  have hc' : (c:ℤ) ≠ 0 := by simp [hc.ne.symm],
+  simp only [modeq_iff_dvd, int.coe_nat_mul, ←mul_sub, mul_dvd_mul_iff_left hc'],
+end
+
+lemma mul_right_iff (hc : 0 < c) : a ≡ b [MOD m] ↔ a * c ≡ b * c [MOD m * c] :=
+begin
+  have hc' : (c:ℤ) ≠ 0 := by simp [hc.ne.symm],
+  simp only [modeq_iff_dvd, int.coe_nat_mul, ←mul_sub_right_distrib, mul_dvd_mul_iff_right hc'],
+end
+
+
+
+-- Apostol, Theorem 5.5
+
+lemma Apostol_5_5' (h : a ≡ b [MOD m]) (hdm : d ∣ m) :
+  ∀ c : ℕ, a ≡ c [MOD d] ↔ b ≡ c [MOD d] :=
+begin
+  have habd := modeq.modeq_of_dvd hdm h,
+  exact λ c, ⟨λ h, (modeq.symm habd).trans h, λ h, modeq.trans habd h⟩,
+end
+
+lemma Apostol_5_5'' (h : a ≡ b [MOD m]) (hdm : d ∣ m) :
+  d ∣ a ↔ d ∣ b :=
+by simpa only [←modeq_zero_iff_dvd] using Apostol_5_5' h hdm 0
+
+
+-- Apostol, Theorem 5.6
+lemma gcd_eq_of_modeq (h : a ≡ b [MOD m]) : gcd a m = gcd b m :=
+begin
+  have h1 := gcd_dvd_right a m,
+  have h2 := gcd_dvd_right b m,
+  exact dvd_antisymm
+    (dvd_gcd ((Apostol_5_5'' h h1).mp (gcd_dvd_left a m)) h1)
+    (dvd_gcd ((Apostol_5_5'' h h2).mpr (gcd_dvd_left b m)) h2),
+end
+
+-- Apostol, Theorem 5.7
+lemma Apostol_5_7 (h : a ≡ b [MOD m]) (h2 : | (b:ℤ) - a | < m) : a = b :=
+begin
+  refine int.coe_nat_inj _,
+  rw modeq_iff_dvd at h,
+  rw eq_comm,
+  rw ←sub_eq_zero,
+  exact int.eq_zero_of_abs_lt_dvd h h2,
+end
+
+
+-- Apostol, Theorem 5.4
+/-- To cancel a common factor `c` from a `modeq` we must divide the modulus `m` by `gcd m c` -/
+lemma Apostol_5_4 (hm : 0 < m) (h : c * a ≡ c * b [MOD m]) :
```

Comment:
Maybe `mul_left_cancel'`? You could also prove a version `mul_left_cancel` where you have `c ≡ d [MOD m]` and `c * a ≡ d * b [MOD m]`, and the corresponding `right` versions.